sm_python/python/sm/__init__.py

```
@@ -83,3 +83,22 @@ def logFatalNamed(name,message):
 
 def logErrorNamed(name,message):
     logNamed(name, LoggingLevel.Error, message, stackdepth=2)
+
+def setExtendableValueStoreValue(evs, key, value, add_if_missing=True):
+    def type2func_string(value):
+        if isinstance(value, str):
+            return "String"
+        elif isinstance(value, bool):
+            return "Bool"
+        elif isinstance(value, int):
+            return "Int"
+        elif isinstance(value, float):
+            return "Double"
```

I would raise a meaningful error if there is now match.